### DISCUSSION TITLE
Contributing to project - how to run web server in local dev environment

### DISCUSSION THREAD

**Discussion by tbarmann** [roleoroleo/yi-hack-Allwinner-v2#839] [General]
Hi,
I'm interested in contributing to this project -- particularly the UI on the webserver. I am wondering how I can get the web server running on a local dev environment to make changes. I see the instructions on setting up a devcontainer, and have been able to do that successfully. But it seems this option is mainly to compile the software to prepare it for a custom firmware release.

I have tried running a simple web server in the `src/www/httpd` directory, but it looks like the `cgi-bin` directory needs to be configured. What webserver daemon is actually running on the yi?  Can this also be run on a local dev environment?

Is there a better way to recreate the web ui locally?

Thanks for any help.

**Comment by roleoroleo** [comment]
I'm not a good web programmer so I'm not able to recommend a dev environment.
But it shouldn't be complicated.

Actually the web server is the little httpd embedded in busybox.
I could help you to build a x64 version to run locally, if you want.
I really need some help on this part.

EDIT

The problem is the cgi part that is stricly related to the hw and cannot be run in a different machine.

**Reply by tbarmann** [reply]
That's helpful to know it's a busybox web server. Yes, I figured the cgi stuff would be an issue. I imagine there has to be a way to set up a dev environment. Perhaps there's a way to set up remote development with vscode connected to the device through ftp or ssh.

**Reply by roleoroleo** [reply]
> Perhaps there's a way to set up remote development with vscode connected to the device through ftp or ssh.

This is be the best way. I will ask to a friend, expert in vsc.

**Comment by 0zLloen** [comment]
Is there any updates in this regard?

**Reply by roleoroleo** [reply]
No, at the moment.
